### PR TITLE
chore(test): add case for validating parent hash

### PR DESCRIPTION
**Pull request by sdaveas**
Add tests to validate block parent hashes. In the light of Flow behaviour explained as per [this](https://interoplabs.slack.com/archives/C021B409SG5/p1724405403164199?thread_ts=1724405178.231209&cid=C021B409SG5) thread.

### CODE REVIEW

File: test/RpcCompatibility.js [hunk 1]
```diff
@@ -243,6 +243,19 @@ describe('RpcCompatibility', () => {
...
             checkBlockTimeStamp(parseInt(block.timestamp, 16), 12000);
         });
+
+        it('blocks have valid parent hashes', async () => {
+            const withTransaction = false;
+            const block = await provider.send('eth_getBlockByNumber', ['latest', withTransaction]);
+            const parentBlock = await provider.send('eth_getBlockByHash', [block.parentHash, withTransaction]);
+
+            console.log('Block:', block);
+            console.log('Parent block:', parentBlock);
```
Comment (milapsheth) [inline test/RpcCompatibility.js:253]: No need to log unless debugging

File: test/RpcCompatibility.js [hunk 2]
```diff
@@ -243,6 +243,19 @@ describe('RpcCompatibility', () => {
             checkBlock(block, false);
             checkBlockTimeStamp(parseInt(block.timestamp, 16), 12000);
         });
+
+        it('blocks have valid parent hashes', async () => {
```
Comment (milapsheth) [inline test/RpcCompatibility.js:247]: Add another test for finalized block (if the chain supports it, that one should work too)

**Review by milapsheth — APPROVED** [review]

File: test/RpcCompatibility.js (hunk 1 above)
Comment (sdaveas) [reply test/RpcCompatibility.js]: oops forgot to cleanup thanks! addressed as part of 75eeb36acf34872bc010efaf682d245903bb553d

File: test/RpcCompatibility.js (hunk 2 above)
Comment (sdaveas) [reply test/RpcCompatibility.js]: sg 75eeb36acf34872bc010efaf682d245903bb553d

File: test/RpcCompatibility.js [hunk 3]
```diff
@@ -39,6 +39,14 @@ describe('RpcCompatibility', () => {
         expect(timeDifference).to.be.lessThanOrEqual(maxDifference);
     }
 
+    async function validParentHashes(tag) {
+        const withTransaction = false;
+        const block = await provider.send('eth_getBlockByNumber', ['latest', withTransaction]);
```
Comment (milapsheth) [inline test/RpcCompatibility.js:44]:
```suggestion
    async function validParentHashes(blockTag) {
        const withTransaction = false;
        const block = await provider.send('eth_getBlockByNumber', [blockTag, withTransaction]);
```

File: test/RpcCompatibility.js [hunk 4]
```diff
@@ -117,11 +125,19 @@ describe('RpcCompatibility', () => {
...
+        describe('supports finalized tag', () => {
+            let finalizedBlockNumber;
 
-            try {
-                const finalizedBlockNumber = await provider.send('eth_getBlockByNumber', ['finalized', false]);
+            before(async () => {
+                try {
+                    finalizedBlockNumber = await provider.send('eth_getBlockByNumber', ['finalized', false]);
+                } catch (error) {
+                    console.error('Failed to retrieve finalized block number:', error);
```
Comment (milapsheth) [inline test/RpcCompatibility.js:135]:
```suggestion
                    console.error('Failed to retrieve finalized block number:', error);
                    throw error;
```
throw again to fail, no point in running the below tests if it's not supported

File: test/RpcCompatibility.js [hunk 5]
```diff
@@ -243,6 +261,10 @@ describe('RpcCompatibility', () => {
             checkBlock(block, false);
             checkBlockTimeStamp(parseInt(block.timestamp, 16), 12000);
         });
+
+        it('should have valid parent hashes', async () => {
+            validParentHashes('latest');
```
Comment (milapsheth) [inline test/RpcCompatibility.js:266]:
```suggestion
            // Note: If chain doesn't have instant finality, reorgs could cause this to fail
            validParentHashes('latest');
```

**Pull request merged**